exercises/rps_pattern_matching.livemd

```diff
@@ -105,7 +110,48 @@ defmodule RockPaperScissorsPatternMatching do
     iex> RockPaperScissorsPatternMatching.play(:scissors, :scissors)
     "draw!"
   """
+
+  # INPUT: 2 guesses (in atom form)
+  # OUTPUT: A string that declares the outcome of the play
+
+  # 5 ideas
+  # Case / Cond
+  # Use With
+
   def play(guess1, guess2) do
+    with true <- guess1 == guess2 do
```

Comment:
Does this work? Remember that Elixir returns the last value in the body of the function, so after each `with` statement the code will keep evaluating.

That means you'll only ever return the value for the last `with` statement.

 with true <-
           {:scissors, :paper} == {guess1, guess2} ||
             {:scissors, :paper} == {guess2, guess1} do
      "Scissors beats paper"
    end
    
    I removed `with` from the current content because I don't think we're ready for It as a class. `with` is used similarly to guards where we protect our code. It's less useful as a logical problem-solving tool. You'd likely be better off relying on `case` or multiple function clauses.